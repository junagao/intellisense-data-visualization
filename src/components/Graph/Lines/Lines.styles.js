import styled from 'styled-components';

const LinePath = styled.path`
  stroke-width: 1.5;
  fill: none;
`;

export default LinePath;
