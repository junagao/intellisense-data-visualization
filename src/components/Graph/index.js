export { default as Axis } from './Axis';
export { default as AxisLabel } from './AxisLabel';
export { default as GraphContainer } from './GraphContainer';
export { default as Lines } from './Lines';
export { default as LinesPoints } from './LinesPoints';
export { default as Tooltip } from './Tooltip';
