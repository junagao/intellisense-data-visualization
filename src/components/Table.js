import React from 'react';
import PropTypes from 'prop-types';

const Table = ({ data }) => {
  const renderTableHeader = () => {
    if (data.length) {
      const header = Object.keys(data[0]);
      return header
        .slice(0, 2)
        .map((key) => (
          <th key={key}>{key.charAt(0).toUpperCase() + key.slice(1)}</th>
        ));
    }
    return null;
  };

  const renderTableData = () =>
    data.map((item) => {
      const { metric, value } = item;
      return value && value[0] !== 'OK' ? (
        <tr key={metric}>
          <td>{metric}</td>
          <td>{value[value.length - 1].toFixed(2)}</td>
        </tr>
      ) : null;
    });

  return (
    <div>
      <table>
        <thead>
          <tr>{renderTableHeader()}</tr>
        </thead>
        <tbody>{renderTableData()}</tbody>
      </table>
    </div>
  );
};

Table.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      metric: PropTypes.string,
      value: PropTypes.arrayOf(
        PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
      ),
    }),
  ).isRequired,
};

export default Table;
