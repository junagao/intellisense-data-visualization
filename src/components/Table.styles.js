import styled from 'styled-components';

export const TableContainer = styled.div`
  min-width: 80%;
`;

export const TableTable = styled.table`
  width: 100%;
  font-size: 0.9rem;
  text-align: left;
  border-radius: 2px 2px 0 0;
  border-collapse: separate;
  border-spacing: 0;
`;

export const TableThead = styled.thead`
  display: table-header-group;
  vertical-align: middle;
  border-top-left-radius: 2px;
  border-top-right-radius: 2px;
`;

export const TableTh = styled.th`
  color: rgba(0, 0, 0, 0.85);
  background-color: #fafafa;
  border-bottom: 1px solid #f0f0f0;
  display: table-cell;
  font-weight: 500;
  text-align: left;
  transition: background 0.3s ease;
  padding: 1rem;
`;

export const TableTr = styled.tr`
  display: table-row;
  position: relative;
  vertical-align: middle;
  &:hover {
    background-color: rgba(0, 0, 0, 0.05);
  }
`;

export const TableTd = styled.td`
  padding: 0.75rem 1rem;
  color: rgba(0, 0, 0, 0.65);
`;
