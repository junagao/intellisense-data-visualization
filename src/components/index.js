export { default as Error } from 'components/Error';
export { default as Loading } from 'components/Loading';
export { default as Table } from 'components/Table';
export {
  Axis,
  AxisLabel,
  GraphContainer,
  Lines,
  LinesPoints,
  Tooltip,
} from 'components/Graph';
