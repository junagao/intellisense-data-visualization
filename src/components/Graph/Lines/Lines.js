import React from 'react';
import PropTypes from 'prop-types';

import { LinesPoints } from 'components';
import LinePath from './Lines.styles';

const Lines = ({ metrics, onHover, transform, scales, marginLeft }) => {
  return metrics.map(({ metric, path, color, dataset }) => (
    <React.Fragment key={metric}>
      <LinePath
        d={path}
        stroke={color}
        transform={transform}
        onMouseOver={() => onHover(metric, color)}
        onFocus={() => onHover(metric, color)}
        onMouseOut={() => onHover(metric, color)}
        onBlur={() => onHover(metric, color)}
      />
      <LinesPoints
        metric={metric}
        dataset={dataset}
        marginLeft={marginLeft}
        color={color}
        scales={scales}
        onHover={onHover}
      />
    </React.Fragment>
  ));
};

Lines.propTypes = {
  metrics: PropTypes.arrayOf(
    PropTypes.shape({
      metric: PropTypes.string,
      path: PropTypes.string,
      color: PropTypes.string,
      dataset: PropTypes.arrayOf(
        PropTypes.shape({
          time: PropTypes.number,
          value: PropTypes.number,
        }),
      ),
    }),
  ).isRequired,
  onHover: PropTypes.func.isRequired,
  transform: PropTypes.string.isRequired,
  scales: PropTypes.shape({
    xScale: PropTypes.func,
    yScale: PropTypes.func,
  }).isRequired,
  marginLeft: PropTypes.number.isRequired,
};

export default Lines;
