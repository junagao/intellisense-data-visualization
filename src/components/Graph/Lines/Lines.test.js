import React from 'react';
import { shallow } from 'enzyme';

import { LinesPoints } from 'components';
import Lines from './Lines';

describe('Lines', () => {
  let wrapper;

  const mockMetrics = [
    {
      metric: 'metric A',
      path: 'M34 3535',
      color: 'red',
      dataset: [
        { time: 0, value: 23 },
        { time: 10, value: 53 },
      ],
    },
  ];
  const mockMarginLeft = 30;
  const mockTransform = `translate(-${mockMarginLeft}, 0)`;
  const mockXScale = jest.fn();
  const mockYScale = jest.fn();
  const mockOnHover = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Lines
        metrics={mockMetrics}
        onHover={mockOnHover}
        transform={mockTransform}
        scales={{ mockXScale, mockYScale }}
        marginLeft={mockMarginLeft}
      />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one Fragment element', () => {
    expect(wrapper.find('Fragment').length).toEqual(1);
  });

  it('should render one LinePath element with prop `d, stroke, transform, onMouseOver, onFocus, onMouseOut, onBlur`', () => {
    expect(wrapper.find('LinePath').length).toEqual(1);
    expect(wrapper.find('LinePath').prop('d')).toEqual(mockMetrics[0].path);
    expect(wrapper.find('LinePath').prop('stroke')).toEqual(
      mockMetrics[0].color,
    );
    expect(wrapper.find('LinePath').prop('transform')).toEqual(mockTransform);
  });

  it('should render one instance of LinesPoints passing the props `metric, dataset, marginLeft, color, scales, onHover`', () => {
    expect(wrapper.find(LinesPoints).length).toEqual(1);
    expect(wrapper.find(LinesPoints).prop('metric')).toEqual('metric A');
    expect(wrapper.find(LinesPoints).prop('dataset')).toEqual(
      mockMetrics[0].dataset,
    );
    expect(wrapper.find(LinesPoints).prop('marginLeft')).toEqual(
      mockMarginLeft,
    );
    expect(wrapper.find(LinesPoints).prop('color')).toEqual(
      mockMetrics[0].color,
    );
    expect(wrapper.find(LinesPoints).prop('scales')).toEqual({
      mockXScale,
      mockYScale,
    });
    expect(wrapper.find(LinesPoints).prop('onHover')).toEqual(mockOnHover);
  });
});
