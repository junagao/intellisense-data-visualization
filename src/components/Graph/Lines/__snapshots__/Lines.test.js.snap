// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lines should render correctly given the required props 1`] = `
<Fragment
  key="91e7490d-9b76-43a4-b7d3-192647683096"
>
  <LinePath
    d="M34 3535"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    stroke="red"
    transform="translate(-30, 0)"
  />
  <LinesPoints
    color="red"
    dataset={
      Array [
        Object {
          "time": 0,
          "value": 23,
        },
        Object {
          "time": 10,
          "value": 53,
        },
      ]
    }
    marginLeft={30}
    metric="metric A"
    onHover={[MockFunction]}
    scales={
      Object {
        "mockXScale": [MockFunction],
        "mockYScale": [MockFunction],
      }
    }
  />
</Fragment>
`;
